server/metrics/metrics.go

```diff
@@ -174,73 +225,95 @@ func NewMetrics(info InstanceInfo) *Metrics {
 	return m
 }
 
-func (m *Metrics) ObserveAPIEndpointDuration(handler, method, statusCode string, elapsed float64) {
+func (m *metrics) GetRegistry() *prometheus.Registry {
+	return m.registry
+}
+
+func (m *metrics) ObserveAPIEndpointDuration(handler, method, statusCode string, elapsed float64) {
 	if m != nil {
 		m.apiTime.With(prometheus.Labels{"handler": handler, "method": method, "status_code": statusCode}).Observe(elapsed)
 	}
 }
 
-func (m *Metrics) ObserveConnectedUsersTotal(count int64) {
+func (m *metrics) ObserveConnectedUsersTotal(count int64) {
 	if m != nil {
 		m.connectedUsersTotal.Set(float64(count))
 	}
 }
 
-func (m *Metrics) ObserveChangeEventTotal(changeType string) {
+func (m *metrics) ObserveChangeEventTotal(changeType string) {
 	if m != nil {
 		m.changeEventTotal.With(prometheus.Labels{"change_type": changeType}).Inc()
 	}
 }
 
-func (m *Metrics) ObserveProcessedChangeEventTotal(changeType string, discardedReason string) {
+func (m *metrics) ObserveProcessedChangeEventTotal(changeType string, discardedReason string) {
 	if m != nil {
 		m.processedChangeEventTotal.With(prometheus.Labels{"change_type": changeType, "discarded_reason": discardedReason}).Inc()
 	}
 }
 
-func (m *Metrics) ObserveLifecycleEventTotal(lifecycleEventType string) {
+func (m *metrics) ObserveLifecycleEventTotal(lifecycleEventType string) {
 	if m != nil {
 		m.lifecycleEventTotal.With(prometheus.Labels{"event_type": lifecycleEventType}).Inc()
 	}
 }
 
-func (m *Metrics) ObserveSyntheticUsersTotal(count int64) {
+func (m *metrics) ObserveMessagesCount(action, source string, isDirectMessage bool) {
+	if m != nil {
+		m.messagesCount.With(prometheus.Labels{"action": action, "source": source, "is_direct": strconv.FormatBool(isDirectMessage)}).Inc()
+	}
+}
+
+func (m *metrics) ObserveReactionsCount(action, source string, isDirectMessage bool) {
+	if m != nil {
+		m.reactionsCount.With(prometheus.Labels{"action": action, "source": source, "is_direct": strconv.FormatBool(isDirectMessage)}).Inc()
+	}
+}
+
+func (m *metrics) ObserveFilesCount(action, source, discardedReason string, isDirectMessage bool, count int64) {
+	if m != nil {
+		m.filesCount.With(prometheus.Labels{"action": action, "source": source, "is_direct": strconv.FormatBool(isDirectMessage), "discarded_reason": discardedReason}).Add(float64(count))
+	}
+}
```

Comment:
I stumbled over the `IncreaseFileCountByOne` pattern -- could we just add a `ObserveFile` and `ObserveFiles(... count)`, with the former doing an `Inc()` and the latter an `Add(count)`?